### ISSUE TITLE
WIN UI 3 Preview 3 XAML parser error using XAMLReader.Load when style is used in the XAML file

### ISSUE THREAD

**Issue by willwhite2** [microsoft/microsoft-ui-xaml#6582]
### Describe the bug

I'm trying to convert a WPF loose XAML application to Win UI 3 preview 3.  I can add the style to the gird and the parsing is successful. If I add a reference to the style I get and XAML parser error. Below is the XAMML file I reading and the code. For  Win UI it looks like a can not use a streamreader as input to the XAMLReader like in WPF. But I found if I read the file, convert it to a string and replace a few characters i can do the read.      Parser has problem when Style="{StaticResource style1}" is added to the XAML file.

      
           string sFileName = CommonDataPathName + "clearwtest.xaml";
            StreamReader mysr = new StreamReader(sFileName);
            string test = mysr.ReadToEnd();
            test = test.Replace("\r\n", " ");
            test = test.Replace("\\", "");
 
            object rootObject = XamlReader.Load(test);

<Grid
xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
 x:Name="GridMain"   >
 <Grid.Resources>
   <Style x:Key="style1" TargetType="RowDefinition">
        <Setter Property="Control.Foreground" Value="Yellow" />
   </Style>
 </Grid.Resources>
<Grid.RowDefinitions>
	<RowDefinition x:Name="Row1" Style="{StaticResource style1}"  Height="1*"/>
<RowDefinition x:Name="Row2"  Height="1*"/>
<RowDefinition x:Name="Row3"  Height="1*"/>
<RowDefinition x:Name="Row4"  Height="1*"/>
<RowDefinition x:Name="Row5"  Height="1*"/>
<RowDefinition x:Name="Row6"  Height="1*"/>
<RowDefinition x:Name="Row7"  Height="1*"/>
<RowDefinition x:Name="Row8"  Height="1*"/>
<RowDefinition x:Name="Row9"  Height="1*"/>
<RowDefinition x:Name="Row10"  Height="1*"/>
<RowDefinition x:Name="Row11"  Height="1*"/>
<RowDefinition x:Name="Row12"  Height="2*"/>

</Grid.RowDefinitions>
<Grid.ColumnDefinitions>
<ColumnDefinition x:Name="DispGridColumn1"  Width="1*"/>
<ColumnDefinition x:Name="DispGridColumn2" Width="1*"/>
<ColumnDefinition x:Name="DispGridColumn3"  Width="1*"/>
<ColumnDefinition x:Name="DispGridColumn4"  Width="1*"/>
<ColumnDefinition x:Name="DispGridColumn5"  Width="1*"/>
<ColumnDefinition x:Name="DispGridColumn6"  Width="1*"/>
<ColumnDefinition x:Name="DispGridColumn7"  Width="1*"/>
<ColumnDefinition x:Name="DispGridColumn8"  Width="1*"/>
</Grid.ColumnDefinitions>
</Grid>

Microsoft.UI.Xaml.Markup.XamlParseException
  HResult=0x802B000A
  Message=XAML parsing failed.
  Source=WinRT.Runtime
  StackTrace:
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr)
   at WinRT.ExceptionHelpers.ThrowExceptionForHR(Int32 hr)
   at ABI.Microsoft.UI.Xaml.Markup.IXamlReaderStatics.global::Microsoft.UI.Xaml.Markup.IXamlReaderStatics.Load(String xaml)
   at Microsoft.UI.Xaml.Markup.XamlReader.Load(String xaml)
   at winui4.MainWindow.testloadingXAML() in C:\Users\whitewi\source\repos\winui4\winui4\MainWindow.xaml.cs:line 137

### Steps to reproduce the bug

See the above code

### Windows app type

- [ ] UWP
- [X] Win32

### Device form factor

Desktop

**Comment by StephenLPeters** [comment]
@RealTommyKlein and @MikeHillberg FYI

**Comment by RealTommyKlein** [comment]
@willwhite2 - does the markup work when placed in a `.xaml` file instead of using `XamlReader.Load`?  Also, what element is setting `Style="{StaticResource style1}"`?  `RowDefinition`s don't have a `Style` property and won't work with `Style`s in general.

**Comment by willwhite2** [comment]
Sorry, this is not a bug. I'm use WPF and assumed similar properties in  Win UI. Changed my loose XAML to use a button with a style instead of a style for the rowdefinition. I created a resource dictionary with the style for the button. read the resource dictionary, added it to the application. Then read the new loose XAML file and added it as a child to a control in the compiled XAML and the controls were rendered.

thanks
